cores/arduino/wiring_analog.c

```diff
@@ -111,11 +111,301 @@ void analogWrite(uint8_t pin, int val)
+				break;
+			#endif
+
+			#if defined(TCCR0A) && defined(COM0B1)
+			case TIMER0B:
+				if(TCCR0A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR1A) && defined(COM1A1)
+			case TIMER1A:
+				if(TCCR1A & (1<<6))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR1A) && defined(COM1B1)
+			case TIMER1B:
+				if(TCCR1A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR1A) && defined(COM1C1)
+			case TIMER1C:
+				if(TCCR1A & (1<<2))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR2) && defined(COM21)
+			case TIMER2:
+				if(TCCR2 & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR2A) && defined(COM2A1)
+			case TIMER2A:
+				if(TCCR2A & (1<<6))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR2A) && defined(COM2B1)
+			case TIMER2B:
+				if(TCCR2A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR3A) && defined(COM3A1)
+			case TIMER3A:
+				if(TCCR3A & (1<<6))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR3A) && defined(COM3B1)
+			case TIMER3B:
+				if(TCCR3A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR3A) && defined(COM3C1)
+			case TIMER3C:
+				if(TCCR3A & (1<<2))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+			#endif
+
+			#if defined(TCCR4A)
+			case TIMER4A:
+				if(TCCR4A & (1<<6))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+			
+			#if defined(TCCR4A) && defined(COM4B1)
+			case TIMER4B:
+				if(TCCR4A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR4A) && defined(COM4C1)
+			case TIMER4C:
+				if(TCCR4A & (1<<2))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+				
+			#if defined(TCCR4C) && defined(COM4D1)
+			case TIMER4D:				
+				if(TCCR4C & (1<<2))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+							
+			#if defined(TCCR5A) && defined(COM5A1)
+			case TIMER5A:
+				if(TCCR5A & (1<<6))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR5A) && defined(COM5B1)
+			case TIMER5B:
+				if(TCCR5A & (1<<4))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			#if defined(TCCR5A) && defined(COM5C1)
+			case TIMER5C: 
+				if(TCCR5A & (1<<2))digitalWrite(pin,HIGH);
+				else digitalWrite(pin,LOW);
+				break;
+			#endif
+
+			case NOT_ON_TIMER:
+			default:
+				if (val < 128) {
+					digitalWrite(pin, LOW);
+				} else {
+					digitalWrite(pin, HIGH);
+				}
```

Comment:
Why the change in style here and lines 403-407? It's a jarring departure from the style convention so thoroughly established by the rest of the code in the switch blocks.

Reply:
That's my mistake. I accidentally left it there.
Sorry, I am new to open source I will fix it asap